src/utils/url-utils.ts

```diff
@@ -0,0 +1,102 @@
+import { useHistory } from 'react-router-dom';
+import moment from 'moment';
+
+export type URLParams = {
+  hazardLayerId?: string;
+  baselineLayerId?: string;
+  date?: number;
+};
+
+/*
+  transforms the string value according to the field specified in the
+  type URLParams
+*/
+const parseValue = (key: string, value: string): URLParams[keyof URLParams] => {
+  switch (key) {
+    case 'date':
+      return moment(value).valueOf();
+    default:
+      return value;
+  }
+};
+
+/*
+  Serialized value to string depending on the parameter specified in the
+  type URLParams
+*/
+const valueToString = (
+  key: string,
+  value: URLParams[keyof URLParams],
+): string => {
+  switch (key) {
+    case 'date':
+      return moment(value).format('YYYY-MM-DD');
+    default:
+      return value as string;
+  }
+};
+
+/*
+  Serialized value to string depending on the parameter specified in the
+  type URLParams
+*/
```

Comment:
Pretty sure there exists existing util to do this for us. Check out `new URLSearchParams`